### ISSUE TITLE
ipld: Shares within a row/col do not preserve original order

### ISSUE THREAD

**Issue by Wondertan** [celestiaorg/celestia-node#627]
The test in https://github.com/celestiaorg/celestia-node/pull/626 uncovered the issue that shares retrieved with `GetLeavesByNamespace` are not ordered in the way they were ordered originally, and we should fix this ASAP to unblock people higher in the stack.

**Comment by Wondertan** [comment]
Duplicate of #621
